src/pairings.rs

```diff
@@ -513,12 +519,33 @@ impl From<G2Affine> for G2Prepared {
     }
 }
 
-#[cfg(feature = "alloc")]
-#[cfg_attr(docsrs, doc(cfg(all(feature = "pairings", feature = "alloc"))))]
+#[cfg_attr(feature = "zeroize", derive(Zeroize))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+// this is implemented as a new type only in order to derive Zeroize
+struct G2Coeffs(Fp2, Fp2, Fp2);
+
+impl G2Coeffs {
+    pub const fn zero() -> Self {
+        Self(Fp2::zero(), Fp2::zero(), Fp2::zero())
+    }
+}
+
+#[cfg(feature = "zeroize")]
+// manual implementation because 'subtle' doesn't have a zeroize feature
+impl Zeroize for G2Prepared {
+    fn zeroize(&mut self) {
+        self.infinity = Choice::from(0);
+        for c in self.coeffs.iter_mut() {
+            c.zeroize();
+        }
+    }
+}
```

Comment:
I do not know whether this results in a valid `G2Prepared`. I think the only way to ensure we get a self-consistent result is to first zeroize the internals, and then replace them with the `G2Prepared` corresponding to `G2Affine::identity()`. We _don't_ want to `impl Default for G2Prepared`, as you should never produce one of these directly.

I'm more inclined to not add zeroization support to `G2Prepared`, or the pairing-specific types generally.

Reply:
I've removed zeroize support in the pairings module except for MillerLoopResult, it seems to make sense there?